### PR TITLE
Improve cm usability

### PR DESCRIPTION
**Pull request by dmitriy-sych**

### CODE REVIEW

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 1]
```diff
@@ -210,4 +212,224 @@
...
+     *         copyManager.copyToDb(ps, sourceInputStream);
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL INSERT Query
+     * @param from              - InputStream with data
+     * @throws SQLException - insertion error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyToDb(ClickHousePreparedStatement preparedStatement, InputStream from) throws SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 2]
```diff
@@ -210,4 +212,224 @@
...
+     *         copyManager.copyToDb(ps, path);
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL INSERT Query
+     * @param path              - Path to the file with data
+     * @throws SQLException - insertion error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyToDb(ClickHousePreparedStatement preparedStatement, Path path) throws IOException, SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 3]
```diff
@@ -210,4 +212,224 @@
...
+     *         copyManager.copyToDb(ps, file);
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL INSERT Query
+     * @param file              - Path to the file with data
+     * @throws SQLException - insertion error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyToDb(ClickHousePreparedStatement preparedStatement, File file) throws IOException, SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 4]
```diff
@@ -210,4 +212,224 @@
...
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL INSERT Query
+     * @param from              - InputStream with data
+     * @param bufferSize        - buffer size for the data transfer
+     * @throws SQLException - insertion error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyToDb(ClickHousePreparedStatement preparedStatement, InputStream from, int bufferSize) throws SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 5]
```diff
@@ -210,4 +212,224 @@
...
+     *         copyManager.copyToDb(ps, sourceReader);
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL INSERT Query
+     * @param from              - Reader with data
+     * @throws SQLException - insertion error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyToDb(ClickHousePreparedStatement preparedStatement, Reader from) throws SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 6]
```diff
@@ -210,4 +212,224 @@
...
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL INSERT Query
+     * @param from              - Reader with data
+     * @param bufferSize        - buffer size for the data transfer
+     * @throws SQLException - insertion error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyToDb(ClickHousePreparedStatement preparedStatement, Reader from, int bufferSize) throws SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 7]
```diff
@@ -210,4 +212,224 @@
...
+     *         copyManager.copyFromDb(ps, writer);
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL and params
+     * @param to                - Writer to write the data to
+     * @throws SQLException - loading error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyFromDb(ClickHousePreparedStatement preparedStatement, Writer to) throws SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java [hunk 8]
```diff
@@ -210,4 +212,224 @@
...
+     *         copyManager.copyFromDb(ps, path);
+     *     }
+     * </pre>
+     *
+     * @param preparedStatement - ClickHousePreparedStatement with SQL and params
+     * @param to                - Path to write data
+     * @throws SQLException - loading error
+     * @see cc.blynk.clickhouse.domain.ClickHouseFormat for possible formats of the data
+     */
+    void copyFromDb(ClickHousePreparedStatement preparedStatement, Path to) throws IOException, SQLException;
```
Comment (doom369) [inline src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: ClickHousePreparedStatement -> PreparedStatement + unwrap()

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 1 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 3 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 4 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 5 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 6 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 7 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 8 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

File: src/main/java/cc/blynk/clickhouse/copy/CopyManager.java (hunk 2 above)
Comment (dmitriy-sych) [reply src/main/java/cc/blynk/clickhouse/copy/CopyManager.java]: done

**Pull request merged**